packages/app/src/elements/items-list.ts

```diff
@@ -52,7 +52,19 @@ function filterByString(fs: string, rec: VaultItem) {
     const content = [rec.name, ...rec.tags, ...rec.fields.map((f) => f.name), ...rec.fields.map((f) => f.value)]
         .join(" ")
         .toLowerCase();
-    return content.search(escapeRegex(fs.toLowerCase())) !== -1;
+    const searchTerms = fs.toLowerCase().split(" ").filter(Boolean);
+
+    if (searchTerms.length <= 1) {
+        return content.search(escapeRegex(fs.toLowerCase())) !== -1;
+    }
+
+    for (const searchTerm of searchTerms) {
+        if (!content.includes(searchTerm)) {
+            return false;
+        }
+    }
+
+    return true;
```

Comment:
I think this should do the same thing in fewer lines (plus it's more readable):

```suggestion
    const searchTerms = escapeRegex(fs.toLowerCase()).split(" ").filter(Boolean);
    
    return searchTerms.every(searchTerm => content.includes(searchTerm));
```

Reply:
Ah, good one!